node/src/transition_frontier/genesis/transition_frontier_genesis_config.rs

```diff
@@ -100,9 +101,14 @@ impl GenesisConfig {
             Self::AccountsBinProt { bytes, constants } => {
                 let mut bytes = bytes.as_ref();
                 let expected_hash = Option::<v2::LedgerHash>::binprot_read(&mut bytes)?;
+                let hashes = Vec::<(u64, v2::LedgerHash)>::binprot_read(&mut bytes)?
+                    .into_iter()
+                    .map(|(idx, hash)| (idx, hash.0.to_field()))
+                    .collect();
                 let accounts = Vec::<ledger::Account>::binprot_read(&mut bytes)?;
 
-                let (mut mask, total_currency) = Self::build_ledger_from_accounts(accounts);
+                let (mut mask, total_currency) =
+                    Self::build_ledger_from_accounts_and_hashes(accounts, hashes);
                 let ledger_hash = ledger_hash(&mut mask);
                 if let Some(expected_hash) = expected_hash.filter(|h| h != &ledger_hash) {
```

Comment:
@binier for the berkeleynet ledger it took about 10 secs here, on mainnet it would take much longer.

Reply:
yeah fairly slow then... What if we cache "merge" hashes, but recompute and compare account hashes? We could also parallelize hashing. 

Any ideas how ocaml node handles this? Is it slow there as well.

Reply:
in OCaml it is just slow (remember they load from the JSON directly), but only an issue the first time, because after the initial load they keep a persisted version of all the important ledgers which include all the relevant data. We are not doing that yet, so this binprot loading we are doing is just a workaround.